### ISSUE TITLE
Controlled LLM Training on Spectral Sphere

### ISSUE THREAD

**Issue by Hiroki11x** [Hiroki11x/Papers#508]
## Links
https://arxiv.org/pdf/2601.08393

<img width="622" height="269" alt="Image" src="https://github.com/user-attachments/assets/6751f7e4-759f-4101-ad25-493818c3c154" />

## 最近の流行り

勾配も、重みもノルムをハードにコントロールして、wd なくす
すると収束や stability, ハイパラの transferability も改善

## TLDR

Spectral Sphere Optimizer (SSO) を提案

- Adam は長い学習で activationが大きくなりやすく muP 満たさなくなる
- Muon は update のノルムが安定するが、weight の変化 (drift) が無視できない

提案

- スケールを不変にするため、重みも更新もスペクトル球面上に制約する、weight decay はやめる
- でも、Megatron でラグランジュを毎回解くのはオーバーヘッド発生 -> いい感じに sharding して pipeline する

結果
- Muon よりも、Adam よりもいい
- muP も改善
